app/main/models/models.py

```diff
@@ -12,6 +12,15 @@
                            db.ForeignKey('sources.id'), primary_key=True)
                  )
 
+# Relationship table to represent the shared users that can access a project
+shared_projects = db.Table('shared_projects',
+                           db.Column('shared_proj', db.Integer,
+                                     db.ForeignKey('projects.id'), primary_key=True),
+                           db.Column('shared_user', db.String),
+                           db.Column('email', db.String, nullable=False),
```

Comment:
I don't think we need to also store the email, since it is inferred from the user id.